waku/waku_rln_relay/rln_relay.nim

```diff
@@ -211,25 +213,30 @@ proc validateMessage*(
   # track message count for metrics
   waku_rln_messages_total.inc()
 
-  # checks if the `msg`'s epoch is far from the current epoch
-  # it corresponds to the validation of rln external nullifier
-  # get current rln epoch
-  let epoch: Epoch = rlnPeer.getCurrentEpoch()
+  # checks if the message's timestamp is within acceptable range
+  let currentTime = Timestamp(getTime().toUnixFloat())
 
-  let
-    msgEpoch = proof.epoch
-    # calculate the gaps
-    gap = absDiff(epoch, msgEpoch)
-
-  trace "epoch info", currentEpoch = fromEpoch(epoch), msgEpoch = fromEpoch(msgEpoch)
-
-  # validate the epoch
-  if gap > rlnPeer.rlnMaxEpochGap:
-    # message's epoch is too old or too ahead
-    # accept messages whose epoch is within +-MaxEpochGap from the current epoch
-    warn "invalid message: epoch gap exceeds a threshold",
-      gap = gap, payloadLen = msg.payload.len, msgEpoch = fromEpoch(proof.epoch)
-    waku_rln_invalid_messages_total.inc(labelValues = ["invalid_epoch"])
+  var timeDiff = uint64(currentTime - msg.timestamp)
+
+  debug "time info",
+    currentTime = currentTime, messageTime = msg.timestamp, msgHash = msg.hash
+
+  if msg.timestamp > currentTime:
+    warn "invalid message: timestamp is in the future", payloadLen = msg.payload.len
+    waku_rln_invalid_messages_total.inc(labelValues = ["future_timestamp"])
+    return MessageValidationResult.Invalid
+
+  if timeDiff > rlnPeer.rlnMaxTimestampGap:
+    warn "invalid message: timestamp difference exceeds threshold",
+      timeDiff = timeDiff, maxTimestampGap = rlnPeer.rlnMaxTimestampGap
+    waku_rln_invalid_messages_total.inc(labelValues = ["invalid_timestamp"])
+    return MessageValidationResult.Invalid
+
+  let computedEpoch = rlnPeer.calcEpoch(msg.timestamp.float64)
+  if proof.epoch != computedEpoch:
+    warn "invalid message: epoch mismatch",
```

Comment:
I think `epoch mismatch` would be ambiguous - the message is not merely mismatching our epoch, its `timestamp` is mismatching it's own epoch. So I would rather say something like: 
`invalid message: timestamp mismatches epoch`